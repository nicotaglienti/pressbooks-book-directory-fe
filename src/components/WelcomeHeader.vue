<template>
  <v-container class="welcome-header">
    <h1>Welcome to Pressbooks Directory</h1>
    <p>
      This directory provides an index of <strong>{{ $store.state.stats.numberOfBooksIndexed }}</strong> books published across
      <strong>{{ $store.state.stats.numberOfNetworksIndexed }}</strong> Pressbooks networks. Learn more about how the directory
      works by <a href="https://networkmanagerguide.pressbooks.com/chapter/how-to-use-the-pressbooks-directory/">reading our guide</a>
      or taking the self-guided tour below.
    </p>
    <v-btn
      class="directory-tour"
      aria-label="Take the tour"
      @click="$store.commit('showTour')"
    >
      Take the tour
    </v-btn>
    <!-- eslint-disable vue/no-v-html -->
    <p
      v-if="additionalText"
      v-html="additionalText"
    />
    <!--eslint-enable-->
    <collections v-if="$vuetify.breakpoint.width > 700" />
    <searchbox />
  </v-container>
</template>

<script>
import Searchbox from './Searchbox';
import Collections from './additionalCards/Collections';

export default {
  name: 'WelcomeHeader',
  components: {Collections, Searchbox},
  data() {
    return {
      additionalText: (process.env.VUE_APP_HEADER_ADDITIONAL_TEXT) ? process.env.VUE_APP_HEADER_ADDITIONAL_TEXT : false,
      totalBooks: false,
      totalNetworks: false
    };
  }
};
</script>