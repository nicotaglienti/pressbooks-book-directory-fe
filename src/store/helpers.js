function setFilters(oldFilters, allowedFilters) {
    let fs = [], ns = [], strQuery = '', f = {};
    let query = [], queryExcluded = [], numerics = {}, q =[];
    let qtyAttr = Object.keys(oldFilters).length;
    for (let attribute in oldFilters) {
        q = [];
        for (let i =0; i < oldFilters[attribute].length; i++) {
            f = oldFilters[attribute][i];
            strQuery = ':';
            if (f.exclude) {
                if (allowedFilters[attribute].type !== 'boolean') {
                    strQuery = ':-';
                }
                queryExcluded.push(f.attribute + strQuery + f.value);
            } else {
                if (f.operator !== undefined) {
                    if (numerics[f.attribute] === undefined) {
                        numerics[f.attribute] = [];
                    }
                    numerics[f.attribute].push(f);
                } else {
                    q.push(f.attribute + strQuery + f.value);
                }
            }
        }
        if (q.length > 0) {
            query.push(q);
        }
    }
    if (queryExcluded.length > 0 && query.length > 0) {
        if (qtyAttr === 1) {
            fs.push(query);
        } else {
            fs = query;
        }
        fs = fs.concat(queryExcluded);
    }
    if (query.length > 0 && queryExcluded.length === 0) {
        fs = fs.concat(query);
    }
    if (query.length === 0 && queryExcluded.length > 0) {
        fs = fs.concat(queryExcluded);
    }
    for (let attr in numerics) {
        if (numerics[attr].length > 1) {
            let v1 = numerics[attr][0].value;
            let v2 = numerics[attr][1].value;
            if (attr === 'storageSize') {
                v1 = numerics[attr][0].value * 1024 * 1024;
                v2 = numerics[attr][1].value * 1024 * 1024;
            }
            ns.push(attr + ':' + v1 + ' TO ' + v2);
        } else {
            let v3 = numerics[attr][0].value;
            if (attr === 'storageSize') {
                v3 = numerics[attr][0].value * 1024*1024;
            }
            strQuery = numerics[attr][0].operator;
            ns.push(attr + strQuery + v3);
        }
    }
    return [fs, ns.join(' AND ')];
}

export default {
    functions: {
        setFilters
    }
};