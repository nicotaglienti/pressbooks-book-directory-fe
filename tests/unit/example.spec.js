import Vuetify from 'vuetify';
import InstantSearch from "vue-instantsearch";

import { store } from "@/store/index";

import WelcomeHeader from '@/components/WelcomeHeader';

import {
  mount,
  createLocalVue
} from '@vue/test-utils';

const localVue = createLocalVue();
localVue.use(Vuetify);
localVue.use(InstantSearch);

describe('WelcomeHeader.vue', () => {
  let vuetify;

  beforeEach(() => {
    vuetify = new Vuetify();
  });

  const wrapper = mount(WelcomeHeader, {
    localVue,
    vuetify,
    store: store
  });

  expect(wrapper.html()).toMatchSnapshot();
  const title = wrapper.find('.headerwelcome__red-font--pressbooks--header');

  expect(title.text()).toBe('Welcome to Pressbooks');
});