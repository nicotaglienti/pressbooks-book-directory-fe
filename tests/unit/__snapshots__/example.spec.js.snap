// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div class="welcome-header">
  <v-container class="welcome-header welcome-header__content">
    <p class="welcome-header welcome-header__content--bold-red">Welcome to Pressbooks Directory</p>
    <p class="welcome-header welcome-header__intro welcome-header__intro--italic">
      This site presents a searchable, filterable directory of <strong>0 </strong>
      public books published across
      <strong>0</strong> Pressbooks networks (see full list [link]).
    </p>
    <p class="welcome-header welcome-header__intro welcome-header__intro--italic">
      You can find books of interest by searching the full text of all metadata, applying one or more filters,
      or simply browsing the book cards below. [More details about using the site.]
    </p>
    <ais-search-box class="searchbox"></ais-search-box>
  </v-container>
</div>
`;
